src/mqtt/MQTT.cpp

```diff
@@ -401,7 +401,7 @@ bool MQTT::wantsLink() const
 
 int32_t MQTT::runOnce()
 {
-    if (!moduleConfig.mqtt.enabled)
+    if (!moduleConfig.mqtt.enabled || !channels.anyMqttEnabled())
```

Comment:
Makes sense indeed to not run the thread in this case, but most of this is already captured by [`MQTT:wantsLink()`](https://github.com/meshtastic/firmware/blob/9c37e57e750a72324ea58f6acda53168c4efae50/src/mqtt/MQTT.cpp#L372), which now also checks whether map reporting is enabled.

Reply:
Just consolidated that channel check with my new method.

Reply:
Should we just use if (!wantsLink()) return disable() perhaps?

Reply:
Changing channel settings does not necessarily cause a reboot, right? If not, it would never re-enter the `runOnce()` if you enable uplink/downlink after enabling MQTT.

Reply:
This is true. We probably need to leave that guard out to be safe.
So this PR will still have some value as moving some of that logic back onto Channel and protecting the client proxy queue from getting spammed for no reason (which I witnessed during testing.

I'll make the changes.

Reply:
Indeed, that part looks good.

Reply:
> Changing channel settings does not necessarily cause a reboot, right? If not, it would never re-enter the `runOnce()` if you enable uplink/downlink after enabling MQTT.

you can always resurrrect a thread by calling `setIntervalFromNow(0)` to get it back out of the MAX_UINT trap. So if we just do that on channel changes we'll be fine.

Reply:
> This is true. We probably need to leave that guard out to be safe. So this PR will still have some value as moving some of that logic back onto Channel and protecting the client proxy queue from getting spammed for no reason (which I witnessed during testing.

See my comment to @GUVWAF :-) we can do this...